operator/internal/controller/vectorized/cluster_controller_test.go

```diff
@@ -0,0 +1,414 @@
+						},
+					},
+				},
+				AdminAPI: []vectorizedv1alpha1.AdminAPI{
+					{
+						Port: 9644,
+						External: vectorizedv1alpha1.ExternalConnectivityConfig{
+							Enabled: false,
+						},
+					},
+				},
+				RPCServer: vectorizedv1alpha1.SocketAddress{
+					Port: 33145,
+				},
+				PandaproxyAPI: []vectorizedv1alpha1.PandaproxyAPI{
+					{
+						Name: "proxy-external",
+						Port: 8082,
+						External: vectorizedv1alpha1.PandaproxyExternalConnectivityConfig{
+							ExternalConnectivityConfig: vectorizedv1alpha1.ExternalConnectivityConfig{
+								Enabled:   true,
+								Subdomain: "test.example.com",
+							},
+							Ingress: &vectorizedv1alpha1.IngressConfig{
+								Enabled:  ptr.To(true),
+								Endpoint: "proxy",
+								Annotations: map[string]string{
+									"nginx.ingress.kubernetes.io/ssl-passthrough": "true",
+								},
+							},
+						},
+					},
+				},
+			},
+			Resources: vectorizedv1alpha1.RedpandaResourceRequirements{
+				ResourceRequirements: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1"),
+						corev1.ResourceMemory: resource.MustParse("2Gi"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1"),
+						corev1.ResourceMemory: resource.MustParse("2Gi"),
+					},
+				},
+			},
+		},
+	}
+
+	require.NoError(t, ctl.Apply(ctx, cluster))
+
+	// Wait for the controller to reconcile and create resources
+	// We don't wait for ClusterConfigured because pods won't become ready in envtest
+	require.NoError(t, ctl.WaitFor(ctx, cluster, func(obj kube.Object, err error) (bool, error) {
+		if err != nil {
+			return false, err
+		}
+
+		cluster := obj.(*vectorizedv1alpha1.Cluster)
+		// Just wait for the controller to have processed the cluster
+		// Check that we have some status conditions, indicating the controller has reconciled
+		return len(cluster.Status.Conditions) > 0, nil
+	}))
+
+	var allClusterResources []client.Object
+	var lists []client.ObjectList
+	for gvk := range testScheme.AllKnownTypes() {
+		if strings.HasSuffix(gvk.Kind, "List") {
+			obj, err := testScheme.New(gvk)
+			require.NoError(t, err)
+
+			if l, ok := obj.(client.ObjectList); ok {
+				lists = append(lists, l)
+			}
+		}
+	}
+
+	// Collect all resources that belong to our cluster
+	clusterSelector := labels.ForCluster(cluster).AsClientSelector()
+	for _, l := range lists {
+		err := ctl.List(ctx, l)
+		if errors.Is(err, &meta.NoKindMatchError{}) {
+			continue
+		}
+		require.NoError(t, err)
+
+		objs, err := meta.ExtractList(l)
+		require.NoError(t, err)
+
+		for _, obj := range objs {
+			o := obj.(client.Object)
+
+			if !clusterSelector.Matches(k8slabels.Set(o.GetLabels())) {
+				continue
+			}
+
+			allClusterResources = append(allClusterResources, o)
+			t.Logf("%T: %s/%s", o, o.GetNamespace(), o.GetName())
+		}
+	}
+
+	// Additionally collect specific resources by name that don't have cluster labels
```

Comment:
I think it would be preferable to go ahead and add the label selectors to these resources in this PR. It's small change that we'll want to make one way or the other.